### DISCUSSION TITLE
Failed to install SingularityCE in WSL

### DISCUSSION THREAD

**Discussion by Kaddal09** [sylabs/singularity#2466] [Q&A]
Hello,

I want to install Singularity on my WSL, I followed the instructions in INSTALL.md, I tried with different version however the outcome still the same. I have installed the dependencies and Go correctly but when I run the command to configure and install Singularity:
`./mconfig
make -C builddir
sudo make -C builddir install`

I got this in the output:
`...
../internal/pkg/runtime/launcher/native/launcher_linux.go:25:2: github.com/sylabs/sif/v2@v2.15.0: Get "https://proxy.golang.org/github.com/sylabs/sif/v2/@v/v2.15.0.zip": tls: failed to verify certificate: x509: certificate signed by unknown authority
../internal/pkg/security/selinux/selinux_supported.go:10:8: github.com/opencontainers/selinux@v1.11.0: Get "https://proxy.golang.org/github.com/opencontainers/selinux/@v/v1.11.0.zip": tls: failed to verify certificate: x509: certificate signed by unknown authority
../pkg/cmdline/env.go:12:2: github.com/spf13/pflag@v1.0.5: Get "https://proxy.golang.org/github.com/spf13/pflag/@v/v1.0.5.zip": tls: failed to verify certificate: x509: certificate signed by unknown authority
make: *** [Makefile:181: singularity] Error 1
make: Leaving directory '/home/kdallah/projects/singularity/builddir'`

AM I missing something to be installed or it´s it because I´m using WSL?

**Comment by dtrudg** [comment]
The message 'certificate signed by unknown authority' shows that there is an issue with the root certificates installed in your WSL environment, or that there is some kind of proxy in your environment that is intercepting the request.

You should try updating your WSL Linux environment first.

Note also that you can install pre-compiled .deb packages from the release page, e.g.: https://github.com/sylabs/singularity/releases/tag/v4.0.2

**Reply by Kaddal09** [reply]
To install the pre-compiled .deb packages, shall I follow [those steps](https://github.com/sylabs/singularity/blob/v4.0.2/INSTALL.md) or it is different?

**Reply by Kaddal09** [reply]
I updated WSL Linux and also the ca-certificates is already the newest version (20230311ubuntu0.22.04.1), and I still got the same issue with certificates, could it be the issue related to the proxy, if so how to check that?

**Reply by dtrudg** [reply]
To install the pre-compiled debs you need to download them, and then use `sudo apt install <filename>` - or if you are in a graphical desktop (e.g. GNOME or KDE) you should be able to double-click them to start the desktop's graphical software installer.

You would have to speak to the administrators of your site to check if there is a transparent / intercepting proxy in place.
